src/backend/distributed/executor/executor.c

```diff
@@ -1609,19 +1611,25 @@ ReceiveResults(WorkerSession *session, bool storeRows)
 		{
 			char *currentAffectedTupleString = PQcmdTuples(result);
 			int64 currentAffectedTupleCount = 0;
+			ShardCommandExecution *shardCommandExecution =
+				session->currentTask->shardCommandExecution;
 
-			if (*currentAffectedTupleString != '\0')
+			/* if there are multiple replicas, make sure to consider only one */
+			if (!shardCommandExecution->gotResults && *currentAffectedTupleString != '\0')
 			{
 				scanint8(currentAffectedTupleString, false, &currentAffectedTupleCount);
 				Assert(currentAffectedTupleCount >= 0);
+
+				execution->rowCount += currentAffectedTupleCount;
 			}
 
-			execution->rowCount += currentAffectedTupleCount;
+			/* no more results */
 			return true;
 		}
 		if (resultStatus == PGRES_TUPLES_OK)
 		{
-			execution->rowCount += PQntuples(result);
+			/* we've already consumed all the tuples, no more results */
+			Assert (PQntuples(result) == 0);
```

Comment:
`DistributedExecution->rowCount` is a bit confusing. The comment in the struct definition says `total number of rows received from shard commands`, which is:
* correct in the case of `SELECT` (for example, if the shard count is 32, then row count will be `32` for `SELECT count(*) from t`), but then how is this useful?
* incorrect in the case of DML. In this case it is number of rows affected.

If it is not useful for `SELECT`s, then what do you think about renaming it to `rowsAffected` or `rowsProcessed`, and updating its comment, and not incrementing it for `SELECT`?

Reply:
With the name/comment updated, it looks good to me.